Test/AwsCommonRuntimeKitTests/http/HTTPClientTestFixture.swift

```diff
@@ -0,0 +1,117 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import XCTest
+@testable import AwsCommonRuntimeKit
+import AwsCHttp
+
+struct HTTPResponse {
+    var statusCode: Int = -1
+    var headers: [HTTPHeader] = [HTTPHeader]()
+    var body: Data = Data()
+}
+
+class HTTPClientTestFixture: XCBaseTestCase {
+    let semaphore = DispatchSemaphore(value: 0)
+
+    func sendHttpRequest(method: String,
+                         endpoint: String,
+                         path: String = "/",
+                         requestBody: String = "",
+                         expectedStatus: Int = 200,
+                         connectionManager: HTTPClientConnectionManager) async throws -> HTTPResponse {
+        var httpResponse = HTTPResponse()
+        let httpRequestOptions = try getHTTPRequestOptions(
+                method: method,
+                endpoint: endpoint,
+                path: path,
+                body: requestBody,
+                response: &httpResponse)
+
+        // Retry the request multiple times to reduce flakiness
+        for i in 1...3 where httpResponse.statusCode != expectedStatus {
```

Comment:
where is flakiness coming from? due to use of external servers?

are all tests using external servers? or there are some that mock server and should not be flaky?

Reply:
Yes, currently, we are not mocking anything and using external servers like `httpbin.org`.

Reply:
lets make a note or update function name to indicate it will do retries? 
it seems a bit dangerous and someone will likely use it in future for cases where test should fail right away without retrying

Reply:
Thanks, I have added a numRetries parameter to make it more configurable.